src/main/java/captain/cybot/adventure/backend/controller/UserController.java

```diff
@@ -0,0 +1,51 @@
+package captain.cybot.adventure.backend.controller;
+
+import captain.cybot.adventure.backend.exception.InvalidRoleException;
+import captain.cybot.adventure.backend.exception.PasswordInvalidException;
+import captain.cybot.adventure.backend.exception.UserAlreadyExistsException;
+import captain.cybot.adventure.backend.model.user.User;
+import captain.cybot.adventure.backend.service.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import javax.validation.Valid;
+import java.net.URI;
+
+@RestController
+@RequestMapping("/api/v0/users")
+@RequiredArgsConstructor
+@Slf4j
+public class UserController {
```

Comment:
Are we having login in user controller or its own service/controller pointing to the same db? If in this class, don't we need a check password call?

Reply:
I know its not obvious at all but, we don't need a separate API for login in the User controller. Basically, we registered the "/api/v0/login" endpoint in our CustomAuthenticationFilter in the SecurityConfig class which performs all the steps necessary to authenticate the user (checks credentials, generates token, etc).